libsolidity/lsp/LanguageServer.h

```diff
@@ -36,6 +36,22 @@ namespace solidity::lsp
 class RenameSymbol;
 enum class ErrorCode;
 
+/**
+ * Enum to mandate what files to take into consideration for source code analysis.
+ */
+enum class FileLoadStrategy
+{
+	/// Takes only those files into consideration that are explicitly opened and those
+	/// that have been directly or indirectly imported.
+	DirectlyOpenedAndOnImported = 0,
+
+	/// Takes all Solidity (.sol) files within the project root into account.
+	/// Symbolic links will be followed, regardless of their destination (no root escalation checks).
```

Comment:
I think it's a good idea to mention allowed paths option by name because it's related. That way this place will come up if you search for it in the code.
```suggestion
	/// Symbolic links will be followed, even if they lead outside of the project directory
	/// (`--allowed-paths` is currently ignored by the LSP).
```